### DISCUSSION TITLE
about the urequests

### DISCUSSION THREAD

**Discussion by QiaoTuCodes** [micropython/micropython#10136] [Libraries & Drivers]
i'm write code :
```python
import requests

url = 'http://127.0.0.1:8000/files'
files = {'file': open('image.jpg', 'rb')}
r = requests.post(url, files=files)
print(r.text)
```
it's work of PC python 3.10.

but try used to MicroPython v1.18 it's not work. why? how can i change that?
```python
import urequests

url = 'http://127.0.0.1:8000/files'
files = {'file': open('image.jpg', 'rb')}
r = urequests.post(url, data=files)
print(r.text)
```

**Comment by karfas** [comment]
> it's not work

seems not to be a very detailed report.

You will need to give us more details.
Where does the micropython run ? On the same server as the working CPython code or on a microcontroller ? 
In the second case, you will most likely need to change the URL to that of your http server.
Any error messages ?

**Reply by QiaoTuCodes** [reply]
The MicroPython v1.18 it's working of esp32-s3 ports. i'll try reproduce PC CPython code method to esp32-s3 port. soon I found the problem is: request.py have files param. the urequests.py not have files param. 
the urequests.py just have data param method. so urequests.post can not transmit dict_class.

**Reply by QiaoTuCodes** [reply]
![image](https://user-images.githubusercontent.com/42764079/204881473-1c8a6545-80e5-4f8c-9d4a-85245ec5f1a6.png)

**Comment by kjm1102** [comment]
https://stackoverflow.com/questions/62423565/how-to-send-image-to-an-api-in-micropython-language could be worth a look

**Comment by GM-Script-Writer-62850** [comment]
You may need to send the image in chunks due to ram limitations, this is probably while files were not implemented in [urequests](https://github.com/micropython/micropython-lib/blob/master/python-ecosys/urequests/urequests.py)
